src/abstractdataframe/join.jl

```diff
@@ -263,7 +270,11 @@ For the three join operations that may introduce missing values (`:outer`, `:lef
 and `:right`), all columns of the returned data table will support missing values.
 
 When merging `on` categorical columns that differ in the ordering of their levels, the
-ordering of the left `DataFrame` takes precedence over the ordering of the right `DataFrame`
+ordering of the left `DataFrame` takes precedence over the ordering of the right `DataFrame`.
+
+If there are more than two data frames passed to `join` the joining is performed
+recursively with left associativity.
```

Comment:
What does this imply in practice? I haven't thought about this too deeply, but I guess it could be more efficient for future optimizations to perform joins in a different order in some cases. I guess that wouldn't change the result as long as we use the same order for columns?

Reply:
I have deliberately restricted this feature to `:inner`, `:outer` and `:cross` to make sure that the only difference would be:
1. efficiency (as you note - but I have left it for later)
2. column names in case `makeunique=true` (different columns might get renamed) - this can be handled in future more efficient implementation, as we have to anyway guarantee predictable column naming result (that is why I specify the contract - this allows the user to know how the columns might get renamed)
3. possibly ordering of rows in `:outer` and `:cross` joins (but again - this can be handled in more efficient implementation, and again - we have to make sure we have a contract here)

On the other hand other kinds of joins either make little sense or would produce different results (e.g. `:right` join would produce different values of rows in the output depending on the order of joins) - that is why I excluded these options (and I think no one really would need them in practice).

Reply:
OK. Any data/thoughts about whether other implementations allow passing several data sets and why?

Reply:
Pandas `join` allows multiple data frames: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.join.html.

In R there is `merge_all` and `merge_recurse` in reshape.

In general I think it is easy to apply `foldl` on a vector of data frames to achieve what is wanted so it was not a top priority to have.
However, I do not see a problem with supporting it, and that is why I thought it is OK to have this PR.

Reply:
OK, cool.